runtime.md

```diff
@@ -34,22 +34,23 @@ This is provided so that consumers can find the container's configuration and ro
 ```
 
 ## Lifecycle
-
-### Create
-
-Creates the container: file system, namespaces, cgroups, capabilities.
-
-### Start (process)
-
-Runs a process in a container.
-Can be invoked several times.
-
-### Stop (process)
-
-Not sure we need that from runc cli.
-Process is killed from the outside.
-
-This event needs to be captured by runc to run onstop event handlers.
+The lifecycle describes the timeline of events that happen from when a container is created to when it ceases to exist.
+
+1. OCI compliant runtime is invoked by passing the bundle path as argument.
+2. The container's runtime environment is created.
+3. The container's state is written to the filesystem.
+4. The prestart hooks are invoked by the runtime.
+   If any prestart hook fails, then the container is killed and the lifecycle continues at step 8.
+5. The user specified process is executed in the container.
+6. The poststart hooks are invoked by the runtime.
+   If any poststart hook fails, then the container is killed and the lifecycle continues at step 8.
+7. Additional actions such as pausing the container, checkpointing the container, resuming the container or signaling the container may be performed using the runtime interface.
```

Comment:
On Tue, Nov 17, 2015 at 05:33:53PM -0800, Vish Kannan wrote:
> > +7. Additional actions such as pausing the container,
> >     checkpointing the container, resuming the container or
> >     signaling the container may be performed using the runtime
> >     interface.
> 
> Checkpointing and restoring is essentially a new phase and it might
> require additional clarification. Shall we exclude those features
> from the initial version?

+1